# Profitable App Profiles for the App Store and Google Play Markets

For this project, the aim is to analyze data of existing apps to be able to understand what type of apps are more likely to attract more users. 

The scope of this analysis is limited to free apps as we assume that the only source of revenue is going to be the in-app ads. This further extends our focus to the user engagement factor with the apps which we will look into. This assumption also directly (and positively) correlates to the number of users on the platform/app. 

We will be looking into both App Store and Google Play markets. 

**Reading the files and deciding on the Sample Data**

Considering the fact that there are over 2 million iOS apps on App Store and 2.1 million Android apps on Google Play. We will break the analysis to a sample of the data . 

In order to save time, the following data sets have been considered:
1. The Google play Data Set : https://dq-content.s3.amazonaws.com/350/googleplaystore.csv
2. The Apple Store Data Set : https://dq-content.s3.amazonaws.com/350/AppleStore.csv

In [1]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In order to be able to extract and use the data efficeintly, the following function  `explore_data()` has been created. This will help us slice the adequate and necessary parts of the lists by asking us to provide the start and end of the list as parameters. 

It is also possible to obtain the row and column length by the means of the function

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line between rows
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


**Google Play Data**

We have 10841 entries in regarding the Google Play applications which are divided over 13 different parameters. 

The parameters which could be relevant are: 'App', 'Category', 'Reviews', 'Installs', 'Type', 'Price', and 'Genres'.

In [3]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


**App Store Data**

We have 7197 entries in regarding the Apple App Store applications which are divided over 16 different parameters.

The parameters which could be relevant are: 'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', and 'prime_genre'

# Data Cleaning

Our scope defines that we need to cater to an Exglish speaking customer segment with apps which are free of cost. 
Hence, the first step would be to clean our data set to be able to make well informed decisions based on accurate data.

**Deleting Wrong Data**

The Google Play data set has a dedicated discussion section, and we can see that one of the discussions outlines an error for row 10472. Let's print this row and compare it against the header and another row that is correct.

Let us first check if the discussion's allegation on the data have any merit by checking the rating (index - 2) for the element in row 10472. Let's print this row and compare it against the header and another row that is correct.


In [4]:
print(android[10472])
print('\n')
print(android_header)
print('\n')
print(android[0][2])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


4.1


The rating for Life Made WI-Fi Touchscreen Photo Frame is 19 which is an error as the max rating is ofcourse 5. Hence we need to delete this data.

In [5]:
print(len(android))
del android[10472] #dont run this more than once as the numbers then get displaced
print(len(android))

10841
10840


Now next: Row 7939 contains NaN instead of 'Free' is the allegation. Let's take a look at that. 

In [6]:
print(android[7939])
print('\n')
print(android_header)

['Curriculum vitae App CV Builder Free Resume Maker', 'BUSINESS', '4.5', '4571', '3.9M', '500,000+', 'Free', '0', 'Everyone', 'Business', 'July 8, 2018', '3.3', '4.1 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


Alas, the allegation is false hence we keep the values untouched. 

**Removing Duplicate Entries**

Part One

If we explore the Google Play data set long enough, we'll find that some apps have more than one entry. For instance, the application Instagram has four entries:

In [7]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In total, there are 1,181 cases where an app occurs more than once:

In [8]:
duplicate_apps = []
unique_apps = []

for row in android:
    name = row[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])
            

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


Now, we certainly do not need duplicate values in our data set as it risks adding a skew to our data due to the positive reinforcement which comes through the repitition of data. 

Hence we would have to remove all the duplicate values and keep each entry in such a way that there is only one unique value. In order to do this, we could just randomly remove the duplicate values. But there is a better way to go about this. 

With the Instagram example presented above, if we notice, the number of reviews is differentin each instance. From the duplicate instances, we can keep the ones with the highest number of ratings as a higher review number tends towards a more reliable data. 

To do that, we will:

1. Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app
2. Use the dictionary to create a new data set, which will have only one entry per app (and we only select the apps with the highest number of reviews)

In [9]:
#Creating a dictionary with the app names and their reviews

reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] > n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

Now we would look at checking if we were able to remove the 1181

In [10]:
print('Expected length:', len(android) - 1181)
print('Actual length:', len(reviews_max))

Expected length: 9659
Actual length: 9659


Now we have identified which ones have the maximum number of reviews and which ones need to be added into the cleaned list. 

We would now have to assign these newly found cleaned data resources into a new list, which we will name `android_clean` 

1. We will creat a list `already_added` to identify the names of the apps which have already been added to the `andoid_clean` variable. 
2. We will run a loop through the original list and append the app name of t which have the number of reviews the same as in `reviews_max` and the names which already do not exist in `already added` (to avoid duplication)
3. For the same condition we would append


In [11]:
already_added = []
android_clean = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name]==n_reviews) and (name not in already_added):
        already_added.append(name)
        android_clean.append(app)

In [12]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9659
Number of columns: 13


** Removing Non English Apps**

We use English for the apps we develop at our company, and we'd like to analyze only the apps that are directed toward an English-speaking audience. However, if we explore the data long enough, we'll find that both data sets have apps with names that suggest they are not directed toward an English-speaking audience.

Considering the ASCII code, if the code is larger than 126 then it is likely that the app has a non-English name. 

Let's first try to write the function we talked about above, and in the next screen we'll remove the rows corresponding to the non-English apps

In [13]:
print(ios[813][1])
print(ios[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
Ultimate Tennis
Dress Up RagazzA13 DX


In [14]:
def is_english(string):
    for char in string:
        if ord(char) > 127:
            return False
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))
      

True
False
False
False


As seen, while in english, when the apps use special characters like smileys (😜) or a trademark sign (™) which would be very common given the brand value attached to the apps, this function returns `'False'`.

However, it is very unlikely to have more than 3 of such special characters in the name of the app. Hence, we would modify our program to return false only if more than three characters have an ASCII of more than 127. 

In [15]:
def is_english(string):
    count = 0
    for char in string:
        if ord(char) > 127:
            count +=1
        if count > 3:
            return False
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
False
True
True


While this is a method, this would only filter out the apps which are not written in the Latin alphabet. There is a possibility that certain German, French, Spanish or Italian apps are still in the list. 

But the data set seems good enough at this point and let us concentrate on how we can go about removing the non-english apps using the function we just created. 

In [16]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)

        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)

explore_data(android_english, 0, 1, True)
print('\n')
explore_data(ios_english, 0 ,1, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


Number of rows: 6183
Number of columns: 16


**Isolating the Free Apps**

Now that we are done cleaning the data, we will look at isolating the Free Apps from the list to further narrow down our search

In [17]:
android_free = []
ios_free = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_free.append(app)

for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_free.append(app)
        
print(len(android_free))
print(len(ios_free))

8862
3222


Now we are left with 8862 Android apps and 3222 iOS (and a general observation that there are more paid apps on App store as compared to Google Play ;) )

**Most Common Apps by Genre**

**Part One**

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we develop it further.
If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.
Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the `prime_genre` column of the App Store data set, and the Genres and Category columns of the Google Play data set.

**Part Two**

We'll build two functions we can use to analyze the frequency tables:

One function to generate frequency tables that show percentages
Another function that we can use to display the percentages in a descending order

In [36]:
def freq_table(data_set, index):
    table = {}
    total = 0

    for app in data_set:
        total += 1
        genre = app[index]
        if genre in table:
            table[genre] +=1
        else:
            table[genre] = 1
            
    table_percentages = {}
    
    for elem in table:
        percentage = (table[elem]/total) * 100
        table_percentages[elem] = percentage
    return table_percentages

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

**Part Three**

We start by examining the frequency table for the prime_genre column of the App Store data set.

In [37]:
display_table(ios_free, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.

Let's continue by examining the Genres and Category columns of the Google Play data set (two columns which seem to be related)

In [38]:
display_table(android_free, 1)

FAMILY : 18.788083953960733
GAME : 9.636650868878357
TOOLS : 8.440532611148726
BUSINESS : 4.581358609794629
LIFESTYLE : 3.9043105393816293
PRODUCTIVITY : 3.8930264048747465
FINANCE : 3.7011961182577298
MEDICAL : 3.5319341006544795
SPORTS : 3.419092755585647
PERSONALIZATION : 3.3175355450236967
COMMUNICATION : 3.2498307379823967
HEALTH_AND_FITNESS : 3.0692845858722633
PHOTOGRAPHY : 2.945159106296547
NEWS_AND_MAGAZINES : 2.798465357707064
SOCIAL : 2.663055743624464
TRAVEL_AND_LOCAL : 2.335815842924848
SHOPPING : 2.2455427668697814
BOOKS_AND_REFERENCE : 2.143985556307831
DATING : 1.8618821936357481
VIDEO_PLAYERS : 1.782893252087565
MAPS_AND_NAVIGATION : 1.399232678853532
EDUCATION : 1.2525389302640486
FOOD_AND_DRINK : 1.2412547957571656
ENTERTAINMENT : 1.0381403746332656
LIBRARIES_AND_DEMO : 0.9365831640713158
AUTO_AND_VEHICLES : 0.9252990295644324
HOUSE_AND_HOME : 0.8350259535093659
WEATHER : 0.8011735499887158
EVENTS : 0.7109004739336493
ART_AND_DESIGN : 0.6770480704129994
PARENTING : 0

The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

In [40]:
display_table(android_free, -4)

Tools : 8.429248476641842
Entertainment : 6.070864364703228
Education : 5.348679756262695
Business : 4.581358609794629
Productivity : 3.8930264048747465
Lifestyle : 3.8930264048747465
Finance : 3.7011961182577298
Medical : 3.5319341006544795
Sports : 3.4642292936131795
Personalization : 3.3175355450236967
Communication : 3.2498307379823967
Action : 3.1031369893929135
Health & Fitness : 3.0692845858722633
Photography : 2.945159106296547
News & Magazines : 2.798465357707064
Social : 2.663055743624464
Travel & Local : 2.324531708417964
Shopping : 2.2455427668697814
Books & Reference : 2.143985556307831
Simulation : 2.0424283457458814
Dating : 1.8618821936357481
Arcade : 1.8618821936357481
Video Players & Editors : 1.782893252087565
Casual : 1.7490408485669149
Maps & Navigation : 1.399232678853532
Food & Drink : 1.2412547957571656
Puzzle : 1.128413450688332
Racing : 0.9930038366057323
Role Playing : 0.9365831640713158
Libraries & Demo : 0.9365831640713158
Auto & Vehicles : 0.92529902956443

Here it is seen that practicality takes the forefront while we are using apps on Google Play where Tools, Education and Business and Productivity are in the top 5 when apps are arranged with respect to genre. 

The android data set has two terms 'Categort (index 1) and Genre (index -4) which are essentially very similar to each other. this could cause some confusion. However, the Genre column seems to be much more granular and intricate which points to a more sophisticated breakdown of the apps. 

We saw which apps are present the most in each of the stores. Now, its time we look at the popularity of the apps and take a look into how many users they have. This is important as we would have to consider a genre with high inflow to get adequate ad-revenue. 

**Most Popular Apps by Genre**

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the `Installs` column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the `rating_count_tot` app.

In [42]:
genres_ios = freq_table(ios_free, -5)

for app in genres_ios:
    total = 0
    len_genre = 0
    for row in ios_free:
        if app == row[-5]:
            n_ratings = float(row[5])
            total += n_ratings
            len_genre += 1
    n_avg_ratings = total / len_genre
    print(app, ':', n_avg_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Finance : 31467.944444444445
Productivity : 21028.410714285714
Entertainment : 14029.830708661417
Navigation : 86090.33333333333
Medical : 612.0
Sports : 23008.898550724636
Food & Drink : 33333.92307692308
Weather : 52279.892857142855
Lifestyle : 16485.764705882353
Health & Fitness : 23298.015384615384
News : 21248.023255813954
Business : 7491.117647058823
Games : 22788.6696905016
Music : 57326.530303030304
Book : 39758.5
Travel : 28243.8
Utilities : 18684.456790123455
Education : 7003.983050847458
Reference : 74942.11111111111
Catalogs : 4004.0
Shopping : 26919.690476190477


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [44]:
for app in ios_free:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


This shows that there are significant players who have a lot of traction, however the new players have little or close to none. There would be a similar pattern for social media platforms as well if we remove the likes of Facebook, Instagram, Twitter and Whatsapp. Another similar pattern could also be detected in Music if we remove Shazam, Spotify, etc which heavily affect and skew our average review ratings. 

We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

Reference apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:

In [45]:
for app in ios_free:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [47]:
for app in ios_free:
    if app[-5] == 'Book':
        print(app[1], ':', app[5])

Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 105274
Color Therapy Adult Coloring Book for Adults : 84062
OverDrive – Library eBooks and Audiobooks : 65450
HOOKED - Chat Stories : 47829
BookShout: Read eBooks & Track Your Reading Goals : 879
Dr. Seuss Treasury — 50 best kids books : 451
Green Riding Hood : 392
Weirdwood Manor : 197
MangaZERO - comic reader : 9
ikouhoushi : 0
MangaTiara - love comic reader : 0
謎解き : 0
謎解き2016 : 0


The book Genre is heavily skewed due to Kindle and Audible both being Amazon based big players. The others do show some potential for example `Hooked` and `Overdrive` which seem to be gaining significant traction. However, the odds do seem to be against considering that the market share is very evidently inequally distributed. 

However considering the fact that dictionary.com was one of the highest reviewed apps in the `Reference` genre section and there are independent apps which also provide ebook/e-Library services within the genre `Book` which are well used by users, a potential idea could be to develop an app with an electronic version of a book, or with quotes, or excerpts with an in-built dictiomnary function. 

Considering that the App Store has many relevant Games and Entertainment related apps, this could be something which the users might consider utilizing. 

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

1. Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

2. Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

3. Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Let's take a look at the Google Play Market

In [49]:
display_table(android_free, 5) # the Installs columns

1,000,000+ : 15.752651771609116
100,000+ : 11.554953735048521
10,000,000+ : 10.505529225908374
10,000+ : 10.223425863236288
1,000+ : 8.406680207628076
100+ : 6.917174452719477
5,000,000+ : 6.8156172421575265
500,000+ : 5.574362446400361
50,000+ : 4.773188896411646
5,000+ : 4.513653802753328
10+ : 3.5432182351613632
500+ : 3.2498307379823967
50,000,000+ : 2.279395170390431
100,000,000+ : 2.1214172872940646
50+ : 1.9183028661701647
5+ : 0.7898894154818324
1+ : 0.5077860528097494
500,000,000+ : 0.2708192281651997
1,000,000,000+ : 0.22568269013766643
0+ : 0.045136538027533285
0 : 0.011284134506883321


In [58]:

categories_android = freq_table(android_free, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_free:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

GAME : 13006872.892271662
NEWS_AND_MAGAZINES : 9549178.467741935
DATING : 854028.8303030303
MAPS_AND_NAVIGATION : 4056941.7741935486
TRAVEL_AND_LOCAL : 13984077.710144928
WEATHER : 5074486.197183099
FOOD_AND_DRINK : 1924897.7363636363
LIFESTYLE : 1437816.2687861272
BEAUTY : 513151.88679245283
EDUCATION : 3057207.207207207
TOOLS : 10695245.286096256
PHOTOGRAPHY : 17805627.643678162
EVENTS : 253542.22222222222
PRODUCTIVITY : 16772838.591304347
PARENTING : 542603.6206896552
FINANCE : 1387692.475609756
BOOKS_AND_REFERENCE : 8767811.894736841
HOUSE_AND_HOME : 1313681.9054054054
ART_AND_DESIGN : 1905351.6666666667
MEDICAL : 107167.23322683707
SOCIAL : 23253652.127118643
FAMILY : 4371709.123123123
SHOPPING : 7036877.311557789
COMMUNICATION : 38326063.197916664
PERSONALIZATION : 5201482.6122448975
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
HEALTH_AND_FITNESS : 4167457.3602941176
ENTERTAINMENT : 19428913.04347826
SPORTS : 4274688.722772277
LIBRARIES_AND_DEMO : 638503.73493

Communication has the highest number of installs but it has to be dominated by players like Whatsapp, Messenger, Gmail etc. Let us dig deep there. 

In [60]:
for app in android_free:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Messenger – Text and Video Chat for Free : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Viber Messenger : 500,000,000+
Firefox Browser fast & private : 100,000,000+
Yahoo Mail – Stay Organized : 100,000,000+
imo beta free calls and text : 100,000,000+
imo free video calls and chat : 500,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Who : 100,000,000+
WeChat : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Android Messages : 100,000,000+
Telegram : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
BBM - Free Calls & Messages : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
WhatsApp Messen

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [62]:
under_100 = []

for app in android_free:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100.append(float(n_installs))
        
sum(under_100) / len(under_100)

3593510.3486590036

A similar pattern can be seen for Videoplayers (mainly due to Youtube, ), Social media and Navigation (as discussed in the ios section)

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

Now. picking up on our discovery in the iOS section. Let us take a look at how the books and reference section fares in this mix. 

In [66]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

Wattpad 📖 Free Books : 100,000,000+
E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Amazon Kindle : 100,000,000+
Cool Reader : 10,000,000+
Dictionary - Merriam-Webster : 10,000,000+
NOOK: Read eBooks & Magazines : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Oxford Dictionary of English : Free : 10,000,000+
Offline: English to Tagalog Dictionary : 500,000+
Spanish English Translator : 10,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
NOOK App for NOOK Devices : 500,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,00

In [67]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Wattpad 📖 Free Books : 100,000,000+
Amazon Kindle : 100,000,000+
Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Audiobooks from Audible : 100,000,000+


Key Observation : These are evidently dominant players with high numbers but are less in quantity. Let us take a look at another bracket of the genre.

In [68]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                      or app[5] == '5,000,000+'
                                      or app[5] == '10,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Dictionary - Merriam-Webster : 10,000,000+
NOOK: Read eBooks & Magazines : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
Oxford Dictionary of English : Free : 10,000,000+
Spanish English Translator : 10,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
English Dictionary - Offline : 10,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) :

Here we can see that the mid-range apps in Google play are more diverse and wide spread as compared to iOS. There are also tonnes of ebook readers and online reading options. This would indicate that there are already many significant players exisiting in the market.

However, there are also many translations in books along with a huge amount of dictionary usage.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

I think Productivity category is also potential. We can create these apps: to do list, reminder, calendar, notepad, habit tracker, and that pomodoro technique app that I mentioned earlier.


# Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.